### PR TITLE
Possible fix for GroupConvolution unit test

### PR DESCRIPTION
**Pull request by yekruglov**
Description: TF Spec: https://www.tensorflow.org/api_docs/python/tf/nn/conv2d
There is no 'group' parameter in Convolution, so MO does not detect it as GroupConvolution if channels in input data shape and filter shape are not equal. TF supports it only on GPU(cudnn)

JIRA: 36578

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests N/A: no new operations were enabled
* [x]  Transformation tests
* [x]  e2e model test with an update of ./tests/e2e_oss/utils/reshape_utils.py N/A: no new models were enabled
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list N/A: no new operations were enabled
* [x]  Supported **public** models list N/A: no new models were enabled
* [x]  New operations specification N/A: no new operations were enabled
* [x]  Guide on how to convert the **public** model N/A: no new models were enabled
* [x]  User guide update N/A

### CODE REVIEW

File: model-optimizer/mo/ops/convolution.py [hunk 1]
```diff
@@ -233,6 +233,8 @@ def infer(node: Node):
         # this attribute should store lambda node: ... (check tf convolution extractor)
         if node.has_valid('get_group'):
             node['group'] = node.get_group(node)
+            if node['group'] == 1 and input_shape[node.channel_dims] != kernel_shape[node.input_feature_channel]:
```
Comment (lazarevevgeny) [inline model-optimizer/mo/ops/convolution.py:236]: @GlebKazantaev , should we implement `get_group` function for TF operations and add this added line to the it?

**Review by lazarevevgeny — COMMENTED** [review]
Comment to Gleb

File: model-optimizer/mo/ops/convolution.py (hunk 1 above)
Comment (lazarevevgeny) [reply model-optimizer/mo/ops/convolution.py:236]: @yekruglov , @GlebKazantaev , any feedback?

Comment (yekruglov) [reply model-optimizer/mo/ops/convolution.py:236]: @lazarevevgeny Maybe we should move it to `extensions/front/tf/conv_ext.py`
Something like `'get_group': lambda node: node.group if 'group' in node and node.group is not None else node.in_node(0).shape[node.channel_dims] / node.kernel_shape[node.input_feature_channel],`

Comment (lazarevevgeny) [reply model-optimizer/mo/ops/convolution.py:236]: Let's do this

File: model-optimizer/extensions/front/tf/conv_ext.py [hunk 2]
```diff
@@ -31,7 +31,8 @@ class Conv2DFrontExtractor(FrontExtractorOp):
     def extract(cls, node):
         attrs = tf_create_attrs(node, 2, 3)
         attrs.update({'op': __class__.op,
-                      'get_group': lambda node: 1,
+                      'get_group': lambda node: node.group if 'group' in node and node.group is not None else
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/front/tf/conv_ext.py:34]: What about `Conv3DFrontExtractor`?

File: model-optimizer/extensions/front/tf/conv_ext.py [hunk 3]
```diff
@@ -31,7 +31,8 @@ class Conv2DFrontExtractor(FrontExtractorOp):
...
+                      node.in_node(0).shape[node.channel_dims] / node.kernel_shape[node.input_feature_channel],
                       'get_output_feature_dim': lambda node: node.kernel_shape[node.output_feature_channel],
                       'get_weights_permute': PermuteAttrs.Permutation(perm=int64_array([3, 2, 0, 1]),
                                                                       inv=int64_array([2, 3, 1, 0]))
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/front/tf/conv_ext.py:38]: Update the PR description according to template.

File: model-optimizer/extensions/front/tf/conv_ext.py [hunk 4]
```diff
@@ -31,7 +31,8 @@ class Conv2DFrontExtractor(FrontExtractorOp):
     def extract(cls, node):
         attrs = tf_create_attrs(node, 2, 3)
         attrs.update({'op': __class__.op,
-                      'get_group': lambda node: 1,
+                      'get_group': lambda node: node.group if 'group' in node and node.group is not None else
+                      node.in_node(0).shape[node.channel_dims] / node.kernel_shape[node.input_feature_channel],
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/front/tf/conv_ext.py:35]: We need layer tests for this change. Please, add new layers to cover this change tests and run existing one.

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

**Comment by lazarevevgeny** [comment]
@yekruglov , do you plan to finish this task?

**Comment by yekruglov** [comment]
> @yekruglov , do you plan to finish this task?

I guess, it is postponed again due to CVS-41517 and CVS-41133

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**